File: NancyDemo.VB/Modules/TreesModule.vb
Imports NancyDemo.VB.Model
Imports Nancy.ModelBinding
Imports NancyDemo.VB.Services
Imports Nancy

Namespace Modules

    Public Class TreesModule
        Inherits NancyModule

        Public Sub New(ByVal treeService As TreeService)
            MyBase.Get("/trees") = Function(parameters)
                                       Return Negotiate.WithModel(New TreesModel() With {.Trees = treeService.AllTrees()})
                                   End Function
            MyBase.Get("/trees/{Id}") = Function(parameters)
                                            Dim result As Integer
                                            Dim isInteger = Integer.TryParse(parameters.id, result)
                                            Dim tree = treeService.FindById(result)
                                            If isInteger AndAlso tree IsNot Nothing Then
                                                Return Negotiate.WithModel(tree)
                                            Else
                                                Return HttpStatusCode.NotFound
                                            End If
                                        End Function
            MyBase.Get("/trees/add/") = Function(parameters)
                                            Return View("AddTree.vbhtml", New TreeModel)
                                        End Function
            Post("/trees/add/") = Function(parameters)
                                      Dim tree = Bind(Of TreeModel)()
                                      treeService.Add(tree)
                                      Return Response.AsRedirect("/trees")
                                  End Function

        End Sub

    End Class
End Namespace

File: NancyDemo.Csharp/Modules/TreesModule.cs
using Nancy.ModelBinding;
using Nancy;
using NancyDemo.Csharp.Model;
using NancyDemo.Csharp.Services;

namespace NancyDemo.Csharp.Modules
{
    public class TreesModule : NancyModule
    {

        public TreesModule(TreeService treeService)
        {
            Get["/trees"] = parameters => Negotiate.WithModel(new TreesModel() { Trees = treeService.AllTrees() });
            Get["/trees/{Id}"] = parameters =>
                {
                    int result;
                    var isInteger = int.TryParse(parameters.id, out result);
                    var tree = treeService.FindById(result);
                    if(isInteger && tree != null)
                    {
                        return Negotiate.WithModel(tree);
                    }
                    else
                    {
                        return HttpStatusCode.NotFound;
                    }
                };
            Get["/trees/add/"] = parameters => View["AddTree.cshtml", new TreeModel()];
            Post["/trees/add/"] = parameters =>
            {
                var tree = this.Bind<TreeModel>();
                treeService.Add(tree);
                return Response.AsRedirect("/trees");
            };
        }
    }
}
